src/helpers/report.cjs

```diff
@@ -1,11 +1,175 @@
+	}
+};
+
+const getReportVersion = (report) => {
+	const { reportVersion } = report;
+
+	switch (reportVersion) {
+		case null:
+		case undefined:
+			throw new Error('Unable to determine report version');
+		default:
+			return reportVersion;
+	}
+};
+
+const injectReportV1Context = (report, context, override) => {
+	const { summary } = report;
+
+	if (!summary) {
+		throw new Error('Report is missing needed property \'summary\'');
+	}
+
+	if (override) {
+		report.summary = {
+			...summary,
+			...flatten(context)
+		};
+	} else if (!validateReportV1ContextAjv(summary)) {
+		report.summary = {
+			...summary,
+			...flatten(context)
+		};
+	}
+
+	return report;
+};
+
+const injectReportContext = (report, context, override) => {
+	const reportVersion = getReportVersion(report);
+
+	switch (reportVersion) {
+		case 1:
+			return injectReportV1Context(report, context, override);
+		default:
+			throw new Error(`Unknown report version '${reportVersion}'`);
 	}
 };
 
-module.exports = { validateReport };
+const injectReportV1LmsInfo = (report, lmsInfo) => {
+	const { summary } = report;
+
+	if (!summary) {
+		throw new Error('Report is missing needed property \'summary\'');
+	}
+
+	const { buildNumber, instanceUrl } = lmsInfo;
+
+	if (buildNumber) {
+		if (!summary.lmsBuildNumber) {
+			summary.lmsBuildNumber = buildNumber;
+		} else {
+			throw new Error('LMS build number already present');
+		}
+	}
+
+	if (instanceUrl) {
+		if (!summary.lmsInstanceUrl) {
+			summary.lmsInstanceUrl = instanceUrl;
+		} else {
+			throw new Error('LMS instance URL already present');
+		}
+	}
+
+	report.summary = summary;
+
+	return report;
+};
+
+const injectReportLmsInfo = (report, lmsInfo) => {
+	const reportVersion = getReportVersion(report);
+
+	switch (reportVersion) {
+		case 1:
+			return injectReportV1LmsInfo(report, lmsInfo);
+		default:
+			throw new Error(`Unknown report version '${reportVersion}'`);
+	}
+};
+
+const upgradeReport = (report) => {
+	const reportVersion = getReportVersion(report);
+
+	switch (reportVersion) {
+		case 1:
+			return report;
+		default:
+			throw new Error(`Unknown report version: ${reportVersion}`);
+	}
+};
+
+class Report {
+	constructor(path, { context, lmsInfo, overrideContext = false } = {}) {
+		let report;
+
+		try {
+			path = resolve(path);
+
+			const contents = fs.readFileSync(path, 'utf8');
+
+			report = JSON.parse(contents);
+		} catch {
+			throw new Error(`Unable to read/parse report at path ${path}`);
+		}
+
+		if (context) {
+			report = injectReportContext(report, context, overrideContext);
+		}
+
+		if (lmsInfo) {
+			report = injectReportLmsInfo(report, context, overrideContext);
+		}
+
+		const reportVersionOriginal = getReportVersion(report);
+
+		validateReport(report, `report (v${reportVersionOriginal})`);
+
+		this._reportVersionOriginal = reportVersionOriginal;
+
+		if (reportVersionOriginal !== latestReportVersionSupported) {
+			upgradeReport(report);
```

Comment:
Right now there is no upgraded version to go to so this will never be run but once we do this will make sure the incoming report is upgraded to the latest version and validated again. Will be useful for downstream tools that rely on the report data. They can just always build toward the latest version and this will take care of the details on upgrading it.